src/main/java/com/redislabs/redisgraph/impl/graph_cache/GraphCacheList.java

```diff
@@ -7,17 +7,16 @@
 import java.util.ArrayList;
 import java.util.List;
 import java.util.concurrent.CopyOnWriteArrayList;
-import java.util.concurrent.locks.ReentrantReadWriteLock;
+
 
 /**
  * Represents a local cache of list of strings. Holds data from a specific procedure, for a specific graph.
  */
-public class GraphCacheList {
+class GraphCacheList {
 
-    private Object mutex = new Object();
+    private final Object mutex = new Object();
```

Comment:
why do you need this mutex? can't you just synchronized on data?

Reply:
changed